Tests/eppo/ConfigurationTests.swift

```diff
@@ -0,0 +1,174 @@
+        eppoClient = EppoClient.initializeOffline(
+            sdkKey: "mock-api-key",
+            assignmentLogger: loggerSpy?.logger,
+            initialConfiguration: try Configuration(
+                flagsConfigurationJson: Data(testJsonString.utf8),
+                obfuscated: false
+            )
+        )
+
+        let configurationObject = eppoClient.getFlagsConfiguration()
+        XCTAssertNotNil(configurationObject, "Flags configuration should not be nil")
+        XCTAssertNotNil(configurationObject?.getFlag(flagKey: "numeric_flag"), "Should contain numeric_flag")
+    }
+
+    // Test initializing a new client with a Configuration object
+    func testInitializeWithConfigurationObject() throws {
+        // First set up initial client
+        eppoClient = EppoClient.initializeOffline(
+            sdkKey: "mock-api-key",
+            assignmentLogger: loggerSpy?.logger,
+            initialConfiguration: try Configuration(
+                flagsConfigurationJson: Data(testJsonString.utf8),
+                obfuscated: false
+            )
+        )
+
+        let configurationObject = eppoClient.getFlagsConfiguration()
+        EppoClient.resetSharedInstance()
+        
+        let newClient = EppoClient.initializeOffline(
+            sdkKey: "mock-api-key",
+            assignmentLogger: loggerSpy?.logger,
+            initialConfiguration: configurationObject
+        )
+        
+        try XCTAssertNotEqual(newClient.getNumericAssignment(
+            flagKey: "numeric_flag",
+            subjectKey: "test-subject",
+            subjectAttributes: [:],
+            defaultValue: 0
+        ), 0)
+    }
+
+    // Test initializing a new client with a JSON string
+    func testInitializeWithJsonString() throws {
+        // First set up initial client
+        eppoClient = EppoClient.initializeOffline(
+            sdkKey: "mock-api-key",
+            assignmentLogger: loggerSpy?.logger,
+            initialConfiguration: try Configuration(
+                flagsConfigurationJson: Data(testJsonString.utf8),
+                obfuscated: false
+            )
+        )
+
+        guard let flagsConfig = eppoClient.getFlagsConfiguration() else {
+            XCTFail("Flags configuration should not be nil")
+            return
+        }
+
+        let configurationString = try flagsConfig.toJsonString()
+        EppoClient.resetSharedInstance()
+        
+        let newClient = EppoClient.initializeOffline(
+            sdkKey: "mock-api-key",
+            assignmentLogger: loggerSpy?.logger,
+            initialConfiguration: try Configuration(
+                flagsConfigurationJson: Data(configurationString.utf8),
+                obfuscated: false
+            )
+        )
+        
+        try XCTAssertNotEqual(newClient.getNumericAssignment(
+            flagKey: "numeric_flag",
+            subjectKey: "test-subject",
+            subjectAttributes: [:],
+            defaultValue: 0
+        ), 0)
```

Comment:
So we are making sure we don't get the default value. But this is deterministic. Should we assert we get a _specific_ value, and that is the same value we get after initializing ~line 107